File: 프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.py
def solution(arr):
    def trans(r):
        return [
            x // 2 if x >= 50 and x % 2 == 0 
            else (x * 2 + 1) if x < 50 and x % 2 == 1 
            else x 
            for x in r
        ]
    cnt = 0
    while True:
        next_arr = trans(arr)
        if arr == next_arr:
            return cnt
        arr = next_arr
        cnt += 1

File: 프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> arr) {
    int count = 0;
    
    while (true) {
        vector<int> next_arr = arr;
        for (int i = 0; i < next_arr.size(); ++i) {
            if (next_arr[i] >= 50 && next_arr[i] % 2 == 0) {
                next_arr[i] /= 2;
            } else if (next_arr[i] < 50 && next_arr[i] % 2 != 0) {
                next_arr[i] = next_arr[i] * 2 + 1;
            }
        }
        
        if (arr == next_arr) {
            return count;
        }
        
        arr = next_arr;
        count++;
    }
}